### ISSUE TITLE
minor issues with layering of secondaries

### ISSUE THREAD

**Issue by tomhughes** [tomhughes/trac-tickets#3781]
**[Submitted to the original trac issue database at 1.37am, Thursday, 26th May 2011]**

Thanks for fixing up layering. There are still a few minor issues that can be seen here: http://www.openstreetmap.org/?lat=28.53707&lon=-81.382107&zoom=18&layers=M

At ground level (South Street) primary and secondary are drawn above motorway_link. But on bridges (Anderson Street) the motorway_link fill is drawn above primary.

Also note how the motorway_link bridge outline covers the edges of the secondary fill. At zoom 17, it looks better, but at zoom 16 the secondary appears completely under the motorway_link and primary. There's also some strangeness at the west end of the bridge with the secondary seeping into the primary, but I don't know if this is fixable.

**Comment by tomhughes** [comment]
**Author: Ldp**
**[Added to the original trac issue at 7.04pm, Thursday, 26th May 2011]**

For non-bridges, the link roads were specifically put before other roads. Even before service, but that's another ticket. :)

For bridges, it's currently a first-come first-served per layer=* number. Tweaked internal ordering should be done for some classes of roads. TODO.

The bleeding bridge outline on Anderson Street is gone after a rerender. It showed on the May 21 tile, but not on todays tile. That was pre-layering change, then.

**Comment by tomhughes** [comment]
**Author: Ldp**
**[Added to the original trac issue at 7.08pm, Thursday, 26th May 2011]**

Let me retract my first-come first-served statement. During the import, osm2pgsql assigns a weight to highway=*, but treats the _link variant at the same weight as the regular road of the same class.

Proper testing will need to be done before the ordering is changed in osm2pgsql. Two drawbacks: will need a full reimport to be applied to all ways, and other osm2pgsql users could be unpleasantly surprised if the order is changed.

Alternative solution could be to ignore the osm2pgsql assigned weight, and assign a weight in the postgresql query.

**Comment by tomhughes** [comment]
**Author: math1985**
**[Added to the original trac issue at 9.15pm, Wednesday, 11th December 2013]**

So the issue is basically: link roads should always be displayed under normal roads, even on tunnels.

**Comment by tomhughes** [comment]
**Author: math1985**
**[Added to the original trac issue at 2.14am, Saturday, 29th March 2014]**

This issue has been resolved in the meanwhile.
